src/Simulation/Simulators.Tests/ResourcesEstimatorTests.cs

```diff
@@ -120,24 +120,40 @@ public void DepthDifferentQubitsTest()
         }
 
         /// <summary>
-        /// Documents that the width and depth statistics reflect independent lower
+        /// Documents that the ExtraWidth and Depth statistics reflect independent lower
         /// bounds for each (two T gates cannot be combined into a circuit of depth
-        /// one and width one).
+        /// one and width one). Width on the other hand is compatible with Depth.
         /// </summary>
         [Fact]
         public void DepthVersusWidthTest()
         {
-            var sim = new ResourcesEstimator();
-
+            // Operation to execute:
             // using(q = Qubit()) { T(q); } using(q = Qubit()) { T(q); } (yes, twice)
-            DepthVersusWidth.Run(sim).Wait();
-            var data = sim.Data;
 
+            // First run with width optimization
+            DataTable data = RunDepthVersusWidthTest(false);
             Assert.Equal(2.0, data.Rows.Find("T")["Sum"]);
+            Assert.Equal(1.0, data.Rows.Find("ExtraWidth")["Sum"]);
             Assert.Equal(1.0, data.Rows.Find("Width")["Sum"]);
+            Assert.Equal(2.0, data.Rows.Find("Depth")["Sum"]);
+
+            // Now run with depth optimization
+            data = RunDepthVersusWidthTest(true);
+            Assert.Equal(2.0, data.Rows.Find("T")["Sum"]);
+            Assert.Equal(1.0, data.Rows.Find("ExtraWidth")["Sum"]);
+            Assert.Equal(2.0, data.Rows.Find("Width")["Sum"]);
             Assert.Equal(1.0, data.Rows.Find("Depth")["Sum"]);
         }
 
+        private DataTable RunDepthVersusWidthTest(bool optimizeDepth) {

```

Comment:
[nit] Minor formatting comment, but generally the convention for C# sources is to have braces on the next line (though Q# and TypeScript still use braces on same line).
```suggestion
        private DataTable RunDepthVersusWidthTest(bool optimizeDepth)
        {

```

Reply:
Updated.